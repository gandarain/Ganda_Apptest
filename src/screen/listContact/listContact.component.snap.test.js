import React from 'react'
import renderer from 'react-test-renderer'

import { mockContacts } from '../../fixture'

import ListContact from './listContact.component'

jest
  .mock('./contact', () => 'Contact')
  .mock('../../component/header', () => 'Header')
  .mock('react-native-config', () => ({
    BASE_URL: ''
  }))
  .mock('../../service', () => ({
    get: () => Promise.resolve(mockContacts)
  }))
  .mock('@react-navigation/native', () => ({
    useNavigation: () => ({
      navigation: () => ({
        navigate: jest.fn()
      })
    })
  }))

describe('List Contact Component', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  it('should render loading', () => {
    const loading = true
    const contacts = []
    const setStateMock = jest.fn()

    React.useState = jest.fn().mockReturnValue([contacts, {}])
    React.useState = jest.fn().mockReturnValue([loading, {}])

    jest
      .spyOn(React, 'useState')
      .mockImplementationOnce(() => [contacts, setStateMock])
      .mockImplementationOnce(() => [loading, setStateMock])

    const rendered = renderer.create(<ListContact />)
    const tree = rendered.toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should render list contact component', () => {
    const loading = false
    const contacts = mockContacts
    const setStateMock = jest.fn()

    React.useState = jest.fn().mockReturnValue([contacts, {}])
    React.useState = jest.fn().mockReturnValue([loading, {}])

    jest
      .spyOn(React, 'useState')
      .mockImplementationOnce(() => [contacts, setStateMock])
      .mockImplementationOnce(() => [loading, setStateMock])

    const rendered = renderer.create(<ListContact />)
    const tree = rendered.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
