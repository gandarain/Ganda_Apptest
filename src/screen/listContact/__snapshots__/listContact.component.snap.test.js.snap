// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Contact Component should render list contact component 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <Header
    right={
      [
        {
          "action": [Function],
          "icon": "plus",
        },
      ]
    }
    title="List Contact"
  />
  <RCTScrollView
    accessibilityLabel="ListContacts"
    data={
      {
        "data": {
          "data": [
            {
              "age": 111,
              "firstName": "Bilbo",
              "id": "93ad6070-c92b-11e8-b02f-cbfa15db428b",
              "lastName": "Baggins",
              "photo": "http://vignette1.wikia.nocookie.net/lotr/images/6/68/Bilbo_baggins.jpg/revision/latest?cb=20130202022550",
            },
          ],
        },
      }
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onRefresh={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={false}
      />
    }
    refreshing={false}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <RCTRefreshControl />
    <View />
  </RCTScrollView>
</View>
`;

exports[`List Contact Component should render loading 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <Header
    right={
      [
        {
          "action": [Function],
          "icon": "plus",
        },
      ]
    }
    title="List Contact"
  />
  <View
    accessibilityLabel="ContainerLoading"
    style={
      {
        "marginVertical": 20,
      }
    }
  >
    <ActivityIndicator
      accessibilityLabel="ActivityIndicator"
      color="#FF6049"
      size="large"
    />
  </View>
</View>
`;

exports[`List Contact Component should render loading when listContactsReducer loading is true 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <Header
    right={
      [
        {
          "action": [Function],
          "icon": "plus",
        },
      ]
    }
    title="List Contact"
  />
  <RCTScrollView
    accessibilityLabel="ListContacts"
    data={
      {
        "data": {
          "data": [
            {
              "age": 111,
              "firstName": "Bilbo",
              "id": "93ad6070-c92b-11e8-b02f-cbfa15db428b",
              "lastName": "Baggins",
              "photo": "http://vignette1.wikia.nocookie.net/lotr/images/6/68/Bilbo_baggins.jpg/revision/latest?cb=20130202022550",
            },
          ],
        },
      }
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onRefresh={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={false}
      />
    }
    refreshing={false}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <RCTRefreshControl />
    <View />
  </RCTScrollView>
</View>
`;
